### PR TITLE
Remove all dependencies on `az`

### PR DESCRIPTION
**Pull request by ellismg**
(This is built on the top of my auth PR which I am landing today, opening this as a draft so I can see a CI run, diff will be much smaller shortly)

### CODE REVIEW

**Review by wbreza — APPROVED** [review]
LGTM

**Review by weikanglim — APPROVED** [review]

**Review by vhvb1989 — APPROVED** [review]
Awesome!

File: cli/Dockerfile [hunk 1]
```diff
@@ -18,8 +18,6 @@ RUN export DEBIAN_FRONTEND=noninteractive \
      && curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | gpg --dearmor -o /usr/share/keyrings/githubcli-archive-keyring.gpg \
      && echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | tee /etc/apt/sources.list.d/github-cli.list > /dev/null \
      && apt-get update && apt-get install -y gh \ 
-     # az cli
-     && curl -sSL https://aka.ms/InstallAzureCLIDeb | bash \
```
Comment (weikanglim) [inline cli/Dockerfile:22]: We should keep this for the Azure DevOps pipelines for now due to #1158

Comment (ellismg) [reply cli/Dockerfile:22]: Good catch, I did not realize that the we ran the the pipeline inside this container.

**Pull request merged**